packages/hash/api/src/graphql/opentelemetry.ts

```diff
@@ -0,0 +1,65 @@
+import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-grpc";
+import { registerInstrumentations } from "@opentelemetry/instrumentation";
+import {
+  ExpressInstrumentation,
+  ExpressLayerType,
+} from "@opentelemetry/instrumentation-express";
+import { GraphQLInstrumentation } from "@opentelemetry/instrumentation-graphql";
+import { HttpInstrumentation } from "@opentelemetry/instrumentation-http";
+import { Resource } from "@opentelemetry/resources";
+import { SimpleSpanProcessor } from "@opentelemetry/sdk-trace-base";
+import { NodeTracerProvider } from "@opentelemetry/sdk-trace-node";
+
+import { logger } from "../logger";
+
+const traceTimeout = 5000;
+
+const unregisterInstrumentations = registerInstrumentations({
+  instrumentations: [
+    new HttpInstrumentation(),
+    new ExpressInstrumentation({
+      ignoreLayersType: [ExpressLayerType.MIDDLEWARE],
+    }),
+    new GraphQLInstrumentation({
+      allowValues: true,
+      depth: 5,
+    }),
```

Comment:
Are these settings based on anything? Do we want to add comments to explain them (e.g. is a depth of `5` a best practice?)

Reply:
The config isn't, the depth was just a best-guess for our deepest queries (our API is quite shallow because of of scalars, so it could probably be less.)

The `allowValues` makes it so values in queries are put into the tracing. We may want to set `sensitive` attributes in the GQL spec to prevent spilling anything that shouldn't be in logs

Reply:
Do you have an idea as to just how much we are logging and instrumenting? I'm wary to add all of this quite intrusive logging without us having any benchmarking in place. Obviously we don't have that infrastructure right now so we should make our best guess, just wondering if you have a sense if we're overdoing it already. (Needing a depth of 5 already makes me quite concerned about the quantity of things we're logging)

Reply:
I have tried to limit amount of logging by the `ignoreLayersType: [ExpressLayerType.MIDDLEWARE],` filter to prevent logging at every single pre/post-request step, so it definitely is a lot less that before, and all of our inner methods are not traced in TypeScript.

As for the depth, that depth of `5` is not needed, and honestly we can probably set it to 1 given our shallow API. I do want to point out that the default is `-1` which is infinite depth, so if anything we're setting an upper limit in this case :)

Reply:
That's a helpful thing to point out, and we can always configure it later. Cheers :)